### PR TITLE
DAOS-4214 test: weekly test error in obj_update_bad_param.py

### PR DESCRIPTION
**Pull request by Puspalogan**
Test-tag: pr,-hw objbadhand objupdatenull invalid_tx single_object
Test-tag: array_object

  - Changed the client code to return -DER_INVAL when akey is NULL
    in iod as opposed to return false.
  - Improved the test code to have container creation in setUp and
    destroy in tearDown.

Signed-off-by: Logan Sundaram <logan.sundaram@intel.com>

### CODE REVIEW

File: src/object/cli_obj.c [hunk 1]
```diff
@@ -1422,7 +1422,7 @@ obj_iod_sgl_valid(unsigned int nr, daos_iod_t *iods, d_sg_list_t *sgls,
 	for (i = 0; i < nr; i++) {
 		if (iods[i].iod_name.iov_buf == NULL)
 			/* XXX checksum & eprs should not be mandatory */
-			return false;
+			return -DER_INVAL;
```
Comment (johannlombardi) [inline src/object/cli_obj.c:1425]: returning a bool is definitely wrong, but i am not sure that return DER_INVAL for all cases is correct as well. @mchaarawi could you please check?

Comment (mchaarawi) [reply src/object/cli_obj.c:1425]: I was the one who suggested this to be fixed this way.
this function is checking all arguments are valid, and in this case the return val is OK to be INVAL incase of an invalid argument. we might want to add more descriptive error codes in the future like INVAL_KEY/TYPE/RECX, etc.

File: src/tests/ftest/object/obj_update_bad_param.py [hunk 2]
```diff
@@ -39,6 +39,23 @@ class ObjUpdateBadParam(TestWithServers):
...
+            self.container.open()
+        except DaosApiError as excep:
+            print(excep)
+            print(traceback.format_exc())
+            self.fail("Test failed during setup .\n")
+
+    def tearDown(self):
+        if self.container:
+            self.container.close()
+            self.container.destroy()
```
Comment (saurabhtandan) [inline src/tests/ftest/object/obj_update_bad_param.py:58]: Don't think this is needed. 
tearDown() in apricot should already be doing that for any container if any.

Comment (Puspalogan) [reply src/tests/ftest/object/obj_update_bad_param.py]: Done.

**Pull request merged**